camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -82,28 +84,69 @@ def generate_show_output(data):
         r"        }}\r\n"
         r"    \],\r\n".format(data["cred_name"])
     )
-    if data.get("exclude_hosts"):
-        output += r'    "exclude_hosts": \[\r\n' r'        "{}"\r\n' r"    \],\r\n".format(
-            data["exclude_hosts"]
+
+    # ─ disable_ssl ─
+    if "disable_ssl" in data:
+        val = str(data["disable_ssl"]).lower()
+    else:
+        val = "null"
+    output += r'    "disable_ssl": {},\r\n'.format(val)
+
+    # ─ exclude_hosts ─
+    if data.get("exclude_hosts") is not None:
+        output += (
+            r'    "exclude_hosts": \[\r\n'
+            r'        "{}"\r\n'
+            r"    \],\r\n".format(data["exclude_hosts"])
         )
-    output += r'    "hosts": \[\r\n' r'        "{}"\r\n' r"    \],\r\n".format(data["hosts"])
-    output += r'    "id": \d+,\r\n'
-    output += r'    "name": "{}",\r\n'.format(data["name"])
-    source_type = data["source_type"]
-    if source_type in QPC_HOST_MANAGER_TYPES:
-        data.setdefault("options", {}).setdefault("ssl_cert_verify", "true")
-    if data.get("options"):
-        output += '    "options": {\r\n'
-        output += ",\r\n".join(
-            [
-                '        "{}": ["]?{}["]?'.format(key, value)
-                for key, value in sorted(data["options"].items(), key=operator.itemgetter(0))
-            ]
+    else:
+        output += r'    "exclude_hosts": null,\r\n'
+
+    # ─ hosts ─
+    output += (
+        r'    "hosts": \[\r\n'
+        r'        "{}"\r\n'
+        r"    \],\r\n".format(data["hosts"])
+    )
+
+    # ─ id and name ─
+    output += (
+        r'    "id": \d+,\r\n'
+        r'    "name": "{}",\r\n'.format(data["name"])
+    )
+
+    # ─ port, proxy, source_type ─
+    output += r'    "port": {},\r\n'.format(data["port"])
+    output += r'    "proxy_url": null,\r\n'
+    output += r'    "source_type": "{}",\r\n'.format(data["source_type"])
+
+    # ─ ssl_cert_verify ─
+    if "ssl_cert_verify" in data:
+        val = data["ssl_cert_verify"]
+        output += r'    "ssl_cert_verify": {},\r\n'.format(
+            str(val).lower() if val is not None else "null"
```

Comment:
That is the only place where you check if `val` is `None`. In other fields, like `ssl_protocol`, you just take dict value, even if `data["ssl_protocol"] = None`.

Is there a reason for special handling of that field?

In general, I would prefer us to be consistent - either always check `val` for None, or never do it. I lean towards "never". If there's a reason for inconsistency, I think it would be good to explain it shortly in a comment.